src/l2_tracker/events.h

```diff
@@ -72,6 +74,63 @@ struct NewServiceNode : L2StateChange {
     static constexpr std::string_view description = "new SN"sv;
 };
 
+struct ContributorV2 {
+    enum class Version {
+        version_invalid,
+        version_0,
+        _count,
+    };
+
+    eth::address address;
+    eth::address beneficiary;
+    uint64_t     amount;
+
+    auto operator<=>(const ContributorV2& o) const = default;
+
+    template <class Archive>
+    void serialize_object(Archive& ar) {
+        auto version = tools::enum_top<Version>;
+        field_varint(ar, "version", version, [](auto v) { return v < Version::_count; });
```

Comment:
Versioning each individual contributor doesn't feel right to me in terms of serialization, because it should be impossible for these values to ever differ from their parent NewServiceNodeV2 serialization's version.

Reply:
... But I can't think of a simple way to make that work here and allow future Contribution modifications, either.

Reply:
Yep you would need a new struct most of the time to handle changes in the structure which would be good to avoid that code explosion.